Lib/test/test_subprocess.py

```diff
@@ -1894,6 +1894,24 @@ def test_extra_groups_error(self):
         with self.assertRaises(ValueError):
             subprocess.check_call([sys.executable, "-c", "pass"], extra_groups=[])
 
+    @unittest.skipUnless(hasattr(os, 'umask'), 'umask() is not available.')
```

Comment:
This test should be skipped in Windows, which implements [`umask`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/umask?view=vs-2019) only to control the default value of the readonly file attribute for files created via `_[w]open` in the current process.

(I know what the Windows CRT does here is nonsense. File attributes are not file permissions. No one can be permitted to write to a readonly file. It's a fundamental attribute of the file that its data cannot be modified. Only `os.access` should take into account the readonly attribute, just as this works for the [i]mmutable file attribute in Linux. We have to live with other people's misinformed design decisions.)

Reply:
Thanks!  Interesting. I didn't realize os.umask was present on Windows but was wondering what would happen there as it'd be a different concept.  Also, this passed the CI regardless... I probably chose a umask that matched what stat returned there?  anyways, skipped on windows now.

Reply:
Currently `os.stat(name).st_mode & 0o666` in Windows will be either 0o666 or 0o444, depending on whether the readonly file attribute is set (see `attributes_to_mode` in "Python/fileutils.c"). For `stat`, the execute bit is also set based on the file extension for file types that are typically executable via `CreateProcessW`: ".exe", ".com", ".bat", and ".cmd". 

This ranges from completely wrong to smelly hack. At some point, real support for file permissions may (should) be implemented, by copying common approaches from other systems such as Cygwin and WSL.